fhirstore/fhirstore.py

```diff
@@ -298,13 +310,32 @@ def search(self, resource_type, params, offset=0, result_size=100, elements=None
         hits = self.es.search(body=query, index=f"fhirstore.{resource_type.lower()}")
         bundle = {
             "resource_type": "Bundle",
-            "items": [h["_source"] for h in hits["hits"]["hits"]],
+            "items": [
+                {"resource": h["_source"], "search": {"mode": "match"}}
+                for h in hits["hits"]["hits"]
+            ],
+
             "total": hits["hits"]["total"]["value"],
         }
 
         if elements:
             bundle["tag"] = {"code": "SUBSETTED"}
 
+        if include:
+            for item in bundle["items"]:
+                for element in include:
+                    included_resource, included_id = re.split(
+                        "\/", item[element]["reference"], maxsplit=1
```

Comment:
add comments to explain what's going on here please

Reply:
Ok